sdk/core/Azure.Core/src/Operation.Rehydration.cs

```diff
@@ -0,0 +1,633 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure;
+
+#pragma warning disable AZC0012 // Avoid single word type names
+public abstract partial class Operation
+#pragma warning restore AZC0012 // Avoid single word type names
+{
+    internal class RehydratedOperation : IOperation
```

Comment:
This isn't a super elegant approach, but provides a temporary fix for AOT warnings introduced by serializing a type defined in another assembly (via internal shared source) to access its internal properties.  Since the type is defined in Azure.Core and instantiated in Azure.Core, as long as the code to implement the Operation.Rehydrate methods lives in Azure.Core all the way down (as opposed to accessing internal shared source that may live in a separate assembly), it can access internal properties for its own internal types without needing to resort to serialization.